libretroshare/src/pqi/authssl.cc

```diff
@@ -989,411 +1002,353 @@ X509 *AuthSSLimpl::SignX509ReqWithGPG(X509_REQ *req, long /*days*/)
  * this is important - as it allows non-friends messages to be validated.
  */
 
-bool AuthSSLimpl::AuthX509WithGPG(X509 *x509,uint32_t& diagnostic)
+bool AuthSSLimpl::checkX509PgpSignature(X509 *x509, uint32_t& diagnostic)
 {
-#ifdef AUTHSSL_DEBUG
-	fprintf(stderr, "AuthSSLimpl::AuthX509WithGPG() called\n");
-#endif
-
-	if (!CheckX509Certificate(x509))
-	{
-		std::cerr << "AuthSSLimpl::AuthX509() X509 NOT authenticated : Certificate failed basic checks" << std::endl;
-		diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_CERTIFICATE_NOT_VALID ;
-		return false;
-	}
-
-	/* extract CN for peer Id */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
-	RsPgpId issuer(std::string(getX509CNString(x509->cert_info->issuer)));
-#else
-	RsPgpId issuer(std::string(getX509CNString(X509_get_issuer_name(x509))));
-#endif
+	RsPgpId issuer = sslcert::getCertIssuer(*x509);
 	RsPeerDetails pd;
-#ifdef AUTHSSL_DEBUG
-	std::cerr << "Checking GPG issuer : " << issuer.toStdString() << std::endl ;
-#endif
-	if (!AuthGPG::getAuthGPG()->getGPGDetails(issuer, pd)) {
-		std::cerr << "AuthSSLimpl::AuthX509() X509 NOT authenticated : AuthGPG::getAuthGPG()->getGPGDetails() returned false." << std::endl;
-		diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_ISSUER_UNKNOWN ;
+	if (!AuthGPG::getAuthGPG()->getGPGDetails(issuer, pd))
+	{
+		RsErr() << __PRETTY_FUNCTION__ << " X509 NOT authenticated : "
+		        << "AuthGPG::getAuthGPG()->getGPGDetails(" << issuer
+		        << ",...) returned false." << std::endl;
+		diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_ISSUER_UNKNOWN;
 		return false;
 	}
+	else
+		Dbg3() << __PRETTY_FUNCTION__ << " issuer: " << issuer << " found"
+		       << std::endl;
 
 	/* verify GPG signature */
-
 	/*** NOW The Manual signing bit (HACKED FROM asn1/a_sign.c) ***/
 
 #if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 	int (*i2d)(X509_CINF*, unsigned char**) = i2d_X509_CINF;
-	ASN1_BIT_STRING *signature = x509->signature;
-	X509_CINF *data = x509->cert_info;
+	ASN1_BIT_STRING* signature = x509->signature;
```

Comment:
WTF?

Reply:
semantically this change nothing, but * actually belong to the type so it got grouped with it, this line change is not strictly needed

Reply:
nope
`int* a, b; ` != `int *a, *b;`

Reply:
exactly. I used to think like you Gio 20 years ago. But sehraf here is right.

Reply:
I know the @sehraf case, and it is a language pitfail...  In this case it semantically change nothing as we are declaring just one variable, and conceptually * is still part of the type, as `signature` is a variable of type `ASN1_BIT_STRING*`.

Reply:
Well, that you use this funny writing in your own code, fine. Although I would not do so myself, I dont really mind. But dont spend time re-writing it in existing code!

Reply:
yeah don't worry I don't plan to do that ;)